Set proper order for default task lists of a project

dleve123 opened the pull request:
# What's in this PR?

This PR changes `TaskList.default_task_lists/0` to return changesets, which means that the `order` of each `TaskList` is now properly set within the scope of a project via `TaskList.changeset/2`.

## References
Fixes #592

dleve123 committed:
Add failing test for expected task lists ordering on project creation

Starts on #592

dleve123 committed:
Manually set the order of the default task lists for a project

dleve123 committed:
Set order of the default task lists through TaskList.changeset/2

* Removes manual and clunky setting of `order` and `position` with
`TaskList.default_task_lists/0`
* Small style refactorings in Project test

Finishes #592

dleve123 commented:
cc: @JoshSmith – I iterated towards what I think is a more robust/idiomatic solution to this bug, but started with the solution you described at https://github.com/code-corps/code-corps-api/pull/622/commits/c66af717835d27d634d1bcfdf29beb7f539a19d3. Let me know what you think of the end solution (d4508b6)

begedin commented on web/models/task_list.ex:
```diff
@@ -29,6 +31,10 @@ defmodule CodeCorps.TaskList do
+    for task_list <- task_list_data do
+      TaskList.changeset(%TaskList{}, task_list)
+    end
   end
```
Just a thought, but this might be nicely chained using

```Elixir
[%{name: "Index", position: 1}, ...]
|> Enum.map(fn (params) -> changeset(%TaskList{}, params) end)
```

dleve123 replied:
@begedin Thanks for the comment! I'm not super familiar with idiomatic Elixir, but seems like a thing is to use a lot of `|>`s, so I'll refactor accordingly and push a commit with that change.

dleve123 committed:
Refactor `TaskList.default_task_lists/0` to use `Enum.map`

Per @begedin's suggestion [in this comment](https://github.com/code-corps/code-corps-api/pull/622?utm_campaign=build-fixed&utm_medium=email&utm_source=notification#discussion_r95074678)

joshsmith commented:
Thanks @dleve123! Would you mind squashing to a single commit? Then we can merge.

joshsmith commented:
@dleve123 just a quick ping here.

joshsmith commented:
Closing in favor of #635 so we can push this up.
